zaak/src/main/kotlin/com/ritense/valtimo/portal/zaak/domain/documenten/Document.kt

```
@@ -32,4 +31,6 @@ data class Document(
     @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
     @GraphQLIgnore
     val status: DocumentStatus?
-)
\ No newline at end of file
+) {
+    val uuid: UUID? get() = url?.let { UUID.fromString(it.substringAfterLast('/')) }
```

If it should always have a UUID, why make it optional?

---

I'm always for fail-fast, so let it crash if it can't work without the UUID. The error should throw some 500 error which can be shown by the UI.

---

I changed it